File: app/src/main/java/com/x7/steambadger/view/adapter/MenuAdapter.java
package com.x7.steambadger.view.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.x7.steambadger.util.AppMenu;

import java.util.ArrayList;
import java.util.List;

public class MenuAdapter extends BaseAdapter {

    private Context context;
    private List<AppMenu> adapter;

    public MenuAdapter(Context context) {
        super();
        this.context = context;
        this.adapter = new ArrayList<AppMenu>();
    }
    
    public void add(AppMenu item) {
		adapter.add(item);
	}
    
    public long getItemId(int position) {
        return position;
    }

    public boolean areAllItemsSelectable() {
        return false;
    }

    @Override
    public boolean isEnabled(int position) {
        return true;
    }
    
    public Object getItem(int position) {
        return adapter.get(position);
    }

    public int getCount() {
        return adapter.size();
    }

    public int getViewTypeCount() {
        return 1;
    }

    public int getItemViewType(int position) {
        return 1;
    }

    public View getView(int position, View convertView, ViewGroup parent) {
    	AppMenu item = adapter.get(position);

        View view = convertView;

        if (view == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = inflater.inflate(android.R.layout.simple_list_item_1, null);
        }

        view.setTag(item.name());

        TextView text = (TextView) view.findViewById(android.R.id.text1);
        text.setText(item.getDescription());
            
        return view;
    }
}

File: app/src/main/java/com/dougmaitelli/steambadger/view/adapter/MenuAdapter.kt
package com.dougmaitelli.steambadger.view.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.dougmaitelli.steambadger.util.AppMenu

import java.util.ArrayList

class MenuAdapter(private val context: Context) : BaseAdapter() {
    private val adapter: MutableList<AppMenu>

    init {
        this.adapter = ArrayList()
    }

    fun add(item: AppMenu) {
        adapter.add(item)
    }

    override fun getCount(): Int {
        return adapter.size
    }

    override fun getViewTypeCount(): Int {
        return 1
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    fun areAllItemsSelectable(): Boolean {
        return false
    }

    override fun isEnabled(position: Int): Boolean {
        return true
    }

    override fun getItem(position: Int): AppMenu {
        return adapter[position]
    }

    override fun getItemViewType(position: Int): Int {
        return 1
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val item = adapter[position]

        var view: View? = convertView

        if (view == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(android.R.layout.simple_list_item_1, null)
        }

        view!!.tag = item.name

        val text = view.findViewById(android.R.id.text1) as TextView
        text.text = item.description

        return view
    }
}